network/transport/nats/nats_connection.go

```diff
@@ -38,6 +38,7 @@ func (c *natsConnection) subscribeToTopics() {
 		} else {
 			c.mutex.Lock()
 			c.natsSubscriptions = append(c.natsSubscriptions, sub)
+			c.mutex.Unlock()
```

Comment:
This was causing some maddening behavior where only the first two topics would get subscribed to, as the loop would block forever on the second call to`c.mutex.Lock`. Since this is happening in a go-routine it was not obvious at all.

Reply:
Very unfortunate. I wonder if we can replace the use of a mutex with go channels.

Reply:
I'm actually not sure if we even need any blocking/locking. The only place we read or modify `natsSubscriptions` is when we subscribe and unsubscribe, so I don't think we'd ever have to worry about concurrent access to the slice.

Reply:
I've removed the mutex in https://github.com/statechannels/go-nitro/pull/1084/commits/d459aba3a5f448a8bb3a97bb81be3c3053a2e193